File: FrontEnd/src/typings/models/empresa.ts
import type { Cliente } from './cliente';
import type { OrdemServico } from './ordemServico';
import type { Profissional } from './profissional';
import type { Ticket } from './ticket';

export interface Empresa {
  Id: number;
  nome: string;
  email: string;
  telefone: string;
  cpf: string;
  endereco: string;
  Cidade: string;
  estado: string;
  cep: string;
  pais: string;
  cnpj?: string;
  profissionalId: number;
  ativo: boolean;
  readonly profissional?: Profissional[];
  clienteId: number;
  readonly cliente?: Cliente[];
  ticketId: number;
  readonly ticket?: Ticket[];
  ordemServicoId: number;
  readonly ordemServico?: OrdemServico[];
}

File: BackEnd/BLL/Models/Empresa.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;

namespace SoftOS.BLL.Models
{
    public class Empresa
    {
        [Key]
        public int Id { get; set; }
        public string Nome { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string Telefone { get; set; } = string.Empty;
        public string? Cpf { get; set; }
        public string Endereco { get; set; } = string.Empty;
        public string Cidade { get; set; } = string.Empty;
        public string Estado { get; set; } = string.Empty;
        public string Cep { get; set; } = string.Empty;
        public string Pais { get; set; } = string.Empty;
        public string? Cnpj { get; set; }
        public bool Ativo { get; set; } = true;
        public virtual ICollection<Profissional>? Profissionais { get; set; }
        public virtual ICollection<Cliente>? Clientes { get; set; }
        public virtual ICollection<Ticket>? Tickets { get; set; }
        public virtual ICollection<OrdemServico>? OrdemServicos { get; set; }
    }
}
